tests/processing/test_harvesting.py

```diff
@@ -270,41 +275,85 @@ def test_scientific_quality_change(self, request_mock, fetch_mock):
             (Mock(status=200), json.dumps(layer()))
         ]
     )
-    def test_layer_harvested_one(self, request_mock, fetch_mock):
-        """Valid layer must be added to database and deleted if not harvested"""
+    def test_hazardset_removed(self, request_mock, fetch_mock):
+        """Empty hazardsets should be removed from database"""
         self.harvester().harvest_layers()
         self.assertEqual(DBSession.query(Layer).count(), 1)
+        self.assertEqual(DBSession.query(HazardSet).count(), 1)
 
         self.harvester().harvest_layers()
         self.assertEqual(DBSession.query(Layer).count(), 0)
+        self.assertEqual(DBSession.query(HazardSet).count(), 0)
 
     @patch.object(
         Harvester,
         "fetch",
         side_effect=[
             layers([{"id": 1}, {"id": 2}]),
-            layers()
+            layers([{"id": 1}])
```

Comment:
Is this needed? Doesn't `layers()` create one layer with id 1?

Reply:
Yes, but more than it works: it works and it's easy to understand.